clusterctl/clusterdeployer/clusterclient.go

```diff
@@ -104,12 +136,33 @@ func (c *clusterClient) Apply(manifest string) error {
 	return c.waitForKubectlApply(manifest)
 }
 
-func (c *clusterClient) GetClusterObjects() ([]*clusterv1.Cluster, error) {
+func (c *clusterClient) GetContextNamespace() string {
+	if c.configOverrides.Context.Namespace == "" {
+		c.configOverrides.Context.Namespace = apiv1.NamespaceDefault
+	}
+	return c.configOverrides.Context.Namespace
+}
+
+func (c *clusterClient) GetClusterObject(name, ns string) (*clusterv1.Cluster, error) {
+	clustersInNamespace, err := c.GetClusterObjectsInNamespace(ns)
+	if err != nil {
+		return nil, err
+	}
+	var cluster *clusterv1.Cluster
+	for _, nc := range clustersInNamespace {
+		if nc.Name == name {
+			cluster = nc
```

Comment:
Is there a reason not to just return here?

Reply:
This was in favor of a single point of return for the method which IMO makes reading the code more easier :)